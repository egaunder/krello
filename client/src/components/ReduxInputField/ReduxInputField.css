.form__group {
  text-align: left; }
  .form__group input[type=text] {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    box-sizing: border-box;
    border-radius: 4px; }
  .form__group input[type=text]::placeholder {
    font-size: 1.5em; }
