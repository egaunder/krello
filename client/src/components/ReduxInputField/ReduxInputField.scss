.form__group {
  text-align: left;
  input[type=text] {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    box-sizing: border-box;
    border-radius: 4px;
  }

  input[type=text]::placeholder {
    font-size: 1.5em;
  }
}