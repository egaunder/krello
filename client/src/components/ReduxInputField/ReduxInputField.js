import React from 'react'
import PropTypes from 'prop-types'
import './ReduxInputField.css'

const ReduxInputField = props => {
  const { input, label, type, placeholder, meta: { touched, error } } = props
  const className = `form__group ${touched && error ? 'form__group--danger' : ''}`
  return (
    <div className={className}>
      <label htmlFor={`${label}`}>{label}</label>
      <input
        className="form-control"
        type={type}
        placeholder={placeholder}
        {...input}
      />
      {touched && (error && <span className="help-block">{error}</span>)}
    </div>
  )
}

ReduxInputField.propTypes = {
  meta: PropTypes.shape({ touched: PropTypes.bool, error: PropTypes.string }).isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  input: PropTypes.shape({}).isRequired,
}

export default ReduxInputField
