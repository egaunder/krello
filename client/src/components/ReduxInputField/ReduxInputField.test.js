import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ReduxInputField from './ReduxInputField'

describe('ReduxInputField', () => {
  const props = {
    input: {},
    placeholder: '',
    type: '',
    meta: {},
    label: '',
  }

  test('should render correctly', () => {
    const output = shallow(<ReduxInputField {...props} />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  test('should contain call form__group', () => {
    const wrapper = mount(<ReduxInputField {...props} />)
    const div = wrapper.find('div')
    expect(div.length).toBe(1)
    expect(div.hasClass('form__group')).toBe(true)
  })
})
