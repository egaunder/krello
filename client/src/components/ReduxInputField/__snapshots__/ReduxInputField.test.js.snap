// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxInputField should render correctly 1`] = `
<div
  className="form__group "
>
  <label
    htmlFor=""
  />
  <input
    className="form-control"
    placeholder=""
    type=""
  />
</div>
`;
